api/src/websocket/utils/items.ts

```diff
@@ -98,20 +60,32 @@ export async function getMultiPayload(
  * @returns the fetched collection data
  */
 export async function getCollectionPayload(
+	subscription: PSubscription,
 	accountability: Accountability | null,
 	schema: SchemaOverview,
 	event?: WebSocketEvent
 ) {
 	const service = new CollectionsService({ schema, accountability });
 
-	if (!event?.action) {
-		return await service.readByQuery();
-	} else if (event.action === 'create') {
-		return await service.readMany([String(event.key)]);
-	} else if (event.action === 'delete') {
-		return event.keys;
-	} else {
-		return await service.readMany(event.keys.map((key: any) => String(key)));
+	if ('item' in subscription) {
+		if (event?.action === 'delete') {
+			// return only the subscribed id in case a bulk delete was done
+			return subscription.item;
+		} else {
+			return await service.readOne(String(subscription.item));
+		}
+	}
+
+	switch (event?.action) {
+		case 'create':
+			return await service.readMany([String(event.key)]);
+		case 'update':
+			return await service.readMany(event.keys.map((key: any) => String(key)));
+		case 'delete':
+			return event.keys;
+		case undefined:
```

Comment:
Is there a special reason for this as it's already covered by `default`?
```suggestion
```

Same on https://github.com/directus/directus/pull/19743/files#diff-bdf71e4ea86237826b20211bbbb174d2bc99964364bd2b334e5e22b21cb32fb8R161

Reply:
Yes `undefined` is an actual value it can be expected to be and `default:` was just added as fallback later for https://github.com/directus/directus/pull/19743#discussion_r1352101638.

Reply:
I see! Not so sure about the `default` then.
Wouldn't it make more sense to handle new event types individually/correctly at the moment when they are added?

TypeScript would warn us if there should be uncovered events:
<img width="358" src="https://github.com/directus/directus/assets/5363448/effcee28-066d-422c-a22e-f78615f526c3">

Reply:
Yep it is still JS in the end so it's not impossible to get other runtime value

Reply:
Okay, works for me 😃 You have the better overview here 👍

Reply:
But i do agree with you, if i had to choose I would drop the `defaults` in favor of explicit values